app/cdap/components/ConnectionList/index.tsx

```diff
@@ -145,42 +158,111 @@ export default function ConnectionList() {
-          setDataForTabsHelper(res, index);
-          toggleLoader(false);
-        });
-      } else {
-        if (entity.canBrowse) {
-          fetchEntities(dataForTabs[1].selectedTab, entity.path as string).then((res) => {
+        fetchEntities(entity.name)
+          .then((res) => {
             setDataForTabsHelper(res, index);
             toggleLoader(false);
+          })
+          .catch((error) => {
+            toggleLoader(false);
+            setIsErrorOnNoWorkSpace(true);
+            // TODO : Need to bind the message on Snackbar . The message is in error.message same as the old UI . Remove the console log later
           });
+      } else {
+        if (entity.canBrowse) {
+          fetchEntities(dataForTabs[1].selectedTab, entity.path as string)
+            .then((res) => {
+              setDataForTabsHelper(res, index);
+              toggleLoader(false);
+            })
+            .catch((error) => {
+              toggleLoader(false);
+              setIsErrorOnNoWorkSpace(true);
+              // TODO : Need to bind the message on Snackbar. Remove the console log later
+            });
         }
       }
       return newData;
     });
   };
 
+  const searchHandler = (index: number) => {
+    setDataForTabs((prev) => {
+      let tempData = [...prev];
+      tempData = tempData.map((eachTempData) => {
+        return {
+          ...eachTempData,
+          isSearching: false,
+        };
+      });
+      tempData[index].isSearching = true;
+      tempData.forEach((eachTempData, innerIndex) => {
+        if (innerIndex === index) {
+          eachTempData.isSearching = true;
+        } else {
+          eachTempData.isSearching = false;
+        }
+      });
+      return tempData;
+    });
+    refs.current[index].focus();
+  };
+
+  const handleSearch = (e: any, index: number) => {
+    const val = e.target.value.toLowerCase();
+    const newData = cloneDeep(dataForTabs);
+    const newDataToSearch = [...newData[index].data];
+    const tempData = newDataToSearch.filter((item: any) => item.name.toLowerCase().includes(val));
+    newData[index].data = [...tempData];
+    setFilteredData(cloneDeep(newData));
+  };
+
+  const handleClearSearch = (e: any, index: number) => {
+    if (refs.current[index].value === '') {
+      setDataForTabs((prev) => {
+        const tempData = [...prev];
+        tempData[index].isSearching = false;
+        return tempData;
+      });
+    } else {
+      refs.current[index].value = '';
+      const newData = cloneDeep(dataForTabs);
+      const newDataToSearch = [...newData[index].data];
+      const tempData = newDataToSearch.filter((item: any) => item.name.toLowerCase().includes(''));
+      newData[index].data = [...tempData];
+      setFilteredData(cloneDeep(newData));
+    }
+  };
+
   useEffect(() => {
     getConnectionsTabData();
   }, []);
 
+  useEffect(() => {
+    const newData = cloneDeep(dataForTabs);
+    setFilteredData(newData);
```

Comment:
can we do setFilteredData( cloneDeep(dataForTabs))

Reply:
Done